### PR TITLE
Add options to enable PSRemoting and register Windows Event Logging Manifest to MSI installer

### LINKED ISSUE #5979: MSI installer run RegisterManifest.ps1 and Enable-PSRemoting
Users may not be aware that they need to run RegisterManifest.ps1 to get logging and Enable-PSRemoting to have a PSCore6 remoting endpoint.  The MSI installer should have two checkboxes: register eventing manifest (checked by default), register powershell remoting (unchecked by default) along with appropriate MSI options for quiet installs.

### PR DESCRIPTION
**Pull request by bergmeister**
## PR Summary

Closes #5979

This is an early working prototype on which I'd like to get feedback on please. It basically re-uses the existing custom dialog and adds 2 new checkboxes to run run `RegisterManifest.ps1` and `Enable-PSRemoting` (with the first one being ticked by default) using the installed pwsh.exe.
Questions to be discussed include e.g. how should pwsh.exe be called and how error should be handled. Currently PowerShell errors are getting ignored by WiX.

## PR Checklist

Note: Please mark anything not applicable to this PR `NA`.

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - [x] Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] User facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - [x] Issue filed - Issue link: https://github.com/PowerShell/PowerShell-Docs/issues/2007
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
    - [feature] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress). I think the recently introduced installer regression should be reverted first before merging this.
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.

### CODE REVIEW

**Comment by iSazonov** [comment]
Scenario - silent installation from command line or aoutomation tools. In the case we should return error if any from this options failed.

**Comment by bergmeister** [comment]
@iSazonov I agree, but would you swallow error when being run via UI? or should the user see only an informational message or the error getting reported as an error? Given that `Enable-PSRemoting` will throw in most scenarios (until we found a solution for it), should there be an additional MSI parameter that determines if errors get suppressed or not?

**Comment by SteveL-MSFT** [comment]
@bergmeister perhaps we can just scope this to Windows Event Logging registration and defer remoting

**Comment by iSazonov** [comment]
Based on my silent scenario I believe UI should write an error (ex.: "Run as Administrator"), block next steps and rollback on pressing `Cancel`.

**Comment by bergmeister** [comment]
This is the final result. I decided to ignore the return value for the moment being because it sounds like you plan to add custom c# actions in the future that would give the possibility of showing a custom error message.
Since the feature provided by the new checkboxes are optional, I think we do not need to require success but I will leave this decision up to you.
image

File: assets/Product.wxs [hunk 1]
```diff
@@ -55,6 +55,31 @@
         shortcut.save
       End If
     </CustomAction>
+
+    <SetProperty Id="RegisterManifest" Before="RegisterManifest" Sequence="execute"
+                 Value ="&quot;[$(var.ProductVersionWithName)]pwsh.exe&quot; -NoProfile -ExecutionPolicy Bypass -File &quot;[$(var.ProductVersionWithName)]RegisterManifest.ps1&quot;" />
```
Comment (SteveL-MSFT) [inline assets/Product.wxs]: Unnecessary whitespace between `Value` and `=`.  Also, I think how you have the attributes on separate lines (61-66) seems more readable.  You don't need to do it for the entire file (could be separate PR), but for the changes on lines 59-60, it would be great to have that done consistently.

File: assets/Product.wxs [hunk 2]
```diff
@@ -213,17 +238,20 @@
...
       <Control Id="Back" Type="PushButton" X="180" Y="243" Width="56" Height="17" Text="!(loc.WixUIBack)"/>
       <Control Id="Next" Type="PushButton" X="236" Y="243" Width="56" Height="17" Cancel="yes" Default="yes" Text="!(loc.WixUINext)"/>
       <Control Id="Cancel" Type="PushButton" X="304" Y="243" Width="56" Height="17" Cancel="yes" Text="!(loc.WixUICancel)">
         <Publish Event="SpawnDialog" Value="CancelDlg">1</Publish>
       </Control>
 
-      <Control Id="Title" Type="Text" X="15" Y="6" Width="200" Height="15" Transparent="yes" NoPrefix="yes" Text="{\WixUI_Font_Title}Explorer Context Menu"/>
-      <Control Id="Description" Type="Text" X="25" Y="23" Width="280" Height="15" Transparent="yes" NoPrefix="yes" Text="Open $(var.ProductName) via right-click in Explorer."/>
+      <Control Id="Title" Type="Text" X="15" Y="6" Width="200" Height="15" Transparent="yes" NoPrefix="yes" Text="{\WixUI_Font_Title}Optional Features"/>
```
Comment (SteveL-MSFT) [inline assets/Product.wxs]: `Optional Features` doesn't seem entirely correct.  Perhaps `Optional Actions`?

**Review by SteveL-MSFT — CHANGES_REQUESTED** [review]

File: assets/Product.wxs (hunk 1 above)
Comment (bergmeister) [reply assets/Product.wxs]: Ok. Done. I was thinking about adding the [XML Tools](https://marketplace.visualstudio.com/items?itemName=DotJoshJohnson.xml) extension of VSCode to the recommended VSCode extensions. This extension would allow us to apply consistent XML formatting.

File: assets/Product.wxs (hunk 2 above)
Comment (bergmeister) [reply assets/Product.wxs]: I don't mind too much about the wording, so I changed it to your suggestion.

File: assets/Product.wxs [hunk 3]
```diff
@@ -55,6 +55,31 @@
...
+                 Value ="&quot;[$(var.ProductVersionWithName)]pwsh.exe&quot; -NoProfile -ExecutionPolicy Bypass -Command &quot;Enable-PSRemoting&quot;" />
+    <CustomAction Id="EnablePSRemoting"
+                  BinaryKey="WixCA"
+                  DllEntry="CAQuietExec"
+                  Execute="deferred"
+                  Return="ignore"
+                  Impersonate="no" />
+    <InstallExecuteSequence>
+      <Custom Action="RegisterManifest" After="InstallFiles">
+        <![CDATA[NOT Installed AND (REGISTER_MANIFEST=1)]]>
```
[the remark this answers is not in the document]
Comment (bergmeister) [reply assets/Product.wxs]: I don't know too much about the details of the script but if you can tell me which script I should call  in which way, we could certainly do that.

[the remark this answers is not in the document]
Comment (iSazonov) [reply assets/Product.wxs]: @SteveL-MSFT @TravisEz13 Any thoughts?

[the remark this answers is not in the document]
Comment (SteveL-MSFT) [reply assets/Product.wxs]: I think we should have a separate issue to update the script (and probably turn it into a cmdlet) to have an uninstall switch that this installer can use.

**Review by SteveL-MSFT — APPROVED** [review]
LGTM

File: assets/Product.wxs (hunk 3 above)
[the remark this answers is not in the document]
Comment (iSazonov) [reply assets/Product.wxs]: @bergmeister Could you please open new tracking Issue?

[the remark this answers is not in the document]
Comment (bergmeister) [reply assets/Product.wxs]: @iSazonov There you go: https://github.com/PowerShell/PowerShell/issues/6215

**Comment by bergmeister** [comment]
@iSazonov I added details about the added MSI properties to the referenced documentation issue but can you also please put the documentation label on this PR please

**Comment by DarwinJS** [comment]
It seems that the way the custom actions are written over these properties means that the configuration cannot be applied during an upgrade due to the condition "Not Installed".  Yet functionally there does not seem to be a reason to prevent them from running during upgrade.

While the following condition is more complex, it should simply limit the actions from not running during "complete uninstall".  It also makes sure the action is not suppressed during an integrated major upgrade uninstall performed by "RemoveExistingProducts"

```
REMOVE ~= “ALL” AND NOT MaintenanceMode ~= “REMOVE” AND NOT UPGRADINGPACKAGECODE
```
The code that these actions would also need to be internally idempotent - not mess up if the changes they make are already present.
